python/pyarrow/pandas_compat.py

```diff
@@ -246,6 +247,11 @@ def construct_metadata(df, column_names, index_levels, index_descriptors,
 def _get_simple_index_descriptor(level):
     string_dtype, extra_metadata = get_extension_dtype_info(level)
     pandas_type = get_logical_type_from_numpy(level)
+    if 'mixed' in pandas_type:
```

Comment:
I'm not familiar, any reason why it's not an equality (`== 'mixed'`) check like the next block? In other words, is the sub-string match desired?

Reply:
It was to catch both 'mixed' and 'mixed-integer', IIRC. But can make a comment (or just write it out explicitly in code)

Reply:
Just wanted to make sure it wasn't a mistake.

Reply:
Added a comment